### ISSUE TITLE
[no-unsafe-argument] conflicts with [no-inferrable-types] in React callback

### ISSUE THREAD

**Issue by HaveSpacesuit** [typescript-eslint/typescript-eslint#4273]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/main/docs/linting/TROUBLESHOOTING.md) and my problem is not listed.

**Repro**

This works as expected:

```ts
const seeSomething = (msg = 'hello'): void => {
  // msg is correctly inferred as `string`
  saySomething(msg);
};


function saySomething(msg: string): void {
  console.log(msg);
}
```

But in a React `useCallback` context, the type of msg is not correctly inferred when trying to pass to the second function.

```ts
function useCaution(): (msg?: string) => void {
  const seeSomething = useCallback((msg = 'hello'): void => {
    // msg incorrectly inferred as `any`
    saySomething(msg); // Unsafe argument of type `any` assigned to a parameter of type `string` (@typescript-eslint-/no-unsafe-argument)
  }, []);

  function saySomething(msg: string): void {
    console.log(msg);
  }

  return seeSomething;
}
```

If I explicitly declare `(msg: string = 'hello')`, I get the error
```ts
// Type string trivially inferred from a string literal, remove type annotation (@typescript-eslint/no-inferrable-types)
```

es lint config
```JSON
{
  "extends": [
    "eslint:recommended",
    "plugin:@typescript-eslint/recommended-requiring-type-checking",
    "plugin:@typescript-eslint/recommended"
  ]
}
```

ts config
```json
{
  "compilerOptions": {
    "target": "es5",
    "module": "commonjs",
    "lib": ["es2019", "dom"],
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "noImplicitReturns": true,
    "allowSyntheticDefaultImports": true,
    "esModuleInterop": true,
    "skipLibCheck": true,
    "resolveJsonModule": true
  }
}
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.5.0` |
| `@typescript-eslint/parser`        | `5.5.0` |
| `TypeScript`                       | `4.5.2` |
| `ESLint`                           | `8.3.0` |
| `node`                             | `12.18.0` |

**Comment by bradzacher** [comment]
Duplicate of #2184, #3145

`no-inferrable-types` does not use type information so it cannot understand the underlying types or understand that the react types define the callback function type as: `TCallback extends (...args: any[]) => any`.

Consider using the `ignoreParameters` option
